tests/common/test_pickle_compat.py

```diff
@@ -58,6 +58,10 @@ def test_pandas_compat(pd_ver: str, name_want: Tuple[str, Callable[[], Any]]) ->
     assert want.equals(got)
 
 
+@pytest.mark.skipif(
+    sys.version_info >= (3, 12, 0),
+    reason="serialization is now controlled by vendored cloudpickle frozen at 2.1.0.",
+)
```

Comment:
Now that we completely control the versions on both ends, we don't need to run these tests for every Python version.